src/bin/radclient.c

```diff
@@ -855,7 +855,7 @@ static int send_one_packet(rc_request_t *request)
 	/*
 	 *	Haven't sent the packet yet.  Initialize it.
 	 */
-	if (request->packet->id == -1) {
+	if (fr_packet_list_find(packet_list, request->packet) == NULL) {
```

Comment:
We should really avoid expensive lookups when a simple flag would do.  It's better to add a flag, or use a flag in `request->packet`.

Reply:
OK.  I thought it didn't matter that much in radclient since the packet_list is short anyway. I'll try to make this cheaper in the next version.

Reply:
I started out adding a new "sent" boolean to the request, but noticed that "tries" already has this info.  Or is there something I'm missing now?  I've updated the branch with a new version